### ISSUE TITLE
upsert does not work for table partition

### ISSUE THREAD

**Issue by xiaoyu-tamu** [jawj/zapatos#74]
`error: virtual tuple table slot does not have system attributes`

```ts
await db
  .upsert('test', [{date:"2021-02-19", sessions:1}], db.constraint('test_pkey'), {
    updateColumns: ['sessions',],
    returning: [],
  })
  .run(pool);
```

```sql
CREATE TABLE test
(
  date     DATE              NOT NULL PRIMARY KEY,
  sessions INTEGER
)
  PARTITION BY RANGE (date);

CREATE TABLE test_default
  PARTITION OF test
    DEFAULT;


// generated sql

INSERT INTO "test" ("date", "sessions")
VALUES ('2021-02-19', 1)
ON CONFLICT ON CONSTRAINT test_pkey DO UPDATE SET ("sessions") = ROW (EXCLUDED."sessions")
RETURNING jsonb_build_object() || jsonb_build_object('$action', CASE xmax WHEN 0 THEN 'INSERT' ELSE 'UPDATE' END) AS result;

```

It works if I remove xmax, maybe add an option to skip it?

**Comment by jawj** [comment]
Hmm, interesting. It seems like the `xmax `hack remains the only way to tell the difference between an INSERT and an UPDATE when upserting, but is not reliable. There's a request to create an official alternative, but no obvious progress.

https://www.postgresql.org/message-id/20190725015448.e5a3rwa22kpnzfe3%40alap3.anarazel.de
https://www.postgresql.org/message-id/1565486215.7551.0%40finefun.com.au

I think you're right that the only solution at present seems to be an option to suppress the `$action` return value. I'll look into that, and I might also enquire on postgres-hackers.

**Comment by jawj** [comment]
https://www.postgresql.org/message-id/DE57F14C-DB96-4F17-9254-AD0AABB3F81F%40mackerron.co.uk

**Comment by jawj** [comment]
Disappointingly, there's no response yet to the Postgres hackers message.

A `reportAction: 'suppress'` option for `upsert` is now available in version 3.5.
